### PR TITLE
Issue 6012: System test for Stream Tags

### LINKED ISSUE #6012: Create System tests for Stream tag APIs
**Describe the solution you'd like**
Create System tests for Stream tag feature.

We need to:
- Create tests for both the GRPC and REST APIs.

### PR DESCRIPTION
**Pull request by shshashwat**
**Change log description**  
Added a new system test for the new feature System Tags.

**Purpose of the change**  
Fixes #6012 

**What the code does**  
Create the streams
Perform tests on empty and non-empty streams
Update streams with tags
Update a stream tag and verify if it is reflected
Seal and Delete the Stream
check if list streams is updated

**How to verify it**  
All tests and system tests must pass

### CODE REVIEW

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java [hunk 1]
```diff
@@ -211,6 +212,50 @@ private void testCreateSealAndDeleteStreams(String scope) {
...
+        final Set<String> tagSet2 = Set.of("t4", "t5", "t6");
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        // Create and Update Streams
+        for (int j = 1; j <= TEST_MAX_STREAMS; j++) {
+            StreamConfiguration config = StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(j)).build();
+            final String stream = "stream"+j;
+            streamManager.createStream(scope, stream, config);
+            streamManager.updateStream(scope, stream, config.toBuilder().tags(tagSet1).build());
+            assertEquals(tagSet1, streamManager.getStreamTags(scope, stream));
+            assertEquals(singletonList(Stream.of(scope, stream)), newArrayList(streamManager.listStreams(scope, "t1")));
```
Comment (shrids) [inline test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java:239]: in the subsequent iterations, the streams returned by `streamManager.listStreams` should not be a singleton list, right ?

**Comment by shshashwat** [comment]
System tests are in progress

**Review by tkaitchuck — APPROVED** [review]

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java [hunk 2]
```diff
@@ -210,6 +215,59 @@ private void testCreateSealAndDeleteStreams(String scope) {
...
+        }
+        // Check the size of streams with tagName t1
+        assertEquals(TEST_MAX_STREAMS, newArrayList(streamManager.listStreams(scope, "t1")).size());
+        // Check if the lists of tag t3 and t1 are equal
+        assertEquals(newArrayList(streamManager.listStreams(scope, "t3")), newArrayList(streamManager.listStreams(scope, "t1")));
+
+        // Update the streams with new tagSet
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        for (int j = 1; j <= TEST_MAX_STREAMS; j++) {
+            StreamConfiguration config = StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(j)).build();
```
Comment (shrids) [inline test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java:249]: nit: We can add the stream tags right here.

**Review by shrids — CHANGES_REQUESTED** [review]
The changes look good. Blocking until we have a successful System test run.

**Comment by shrids** [comment]
The test successful ran on `test-pravega-continuous-1/1670/`

**Review by shrids — APPROVED** [review]

**Pull request merged**